src/main/scala/mesosphere/marathon/state/Migration.scala

```diff
@@ -364,9 +364,20 @@ class MigrationTo0_13(taskRepository: TaskRepository, store: PersistentStore) {
         _ <- store.delete(oldName)
       } yield ()
     }
-    store.load(oldName).flatMap {
-      case Some(key) => moveKey(key.bytes)
-      case None      => Future.successful(())
+
+    store.load(newName).flatMap {
+      case Some(_) =>
+        log.info("framework:id already exists, no need to migrate")
+        Future.successful(())
```

Comment:
Exactly. I've thought about removing it, but its a corner case that's merely irritating if someone looks directly into zk.

If the `renameFrameworkId` crashes while deleting the old key (zk timeout or whatever), we _can_ end up here. Then the migration will be tried again, skips this step because the new key exists, and will finish successfully.